stream_alert/apps/app_base.py

```diff
@@ -335,7 +336,15 @@ def _make_post_request(self, full_url, headers, data, is_json=True):
             response = requests.post(full_url, headers=headers,
                                      data=data, timeout=self._DEFAULT_REQUEST_TIMEOUT)
 
-        return self._check_http_response(response), response.json()
+        try:
+            return self._check_http_response(response), response.json()
+        except JSONDecodeError:
+            # https://github.com/airbnb/streamalert/issues/998
+            # When response returns Gateway Timeout with status_code 504, the response
+            # object will return empty string and raises JSONDecoderError a when .json() refers to.
+            # See https://github.com/psf/requests/blob/v2.22.0/requests/models.py#L853
+            # Instead of raising exception, we can just return False, None
+            return False, None
```

Comment:
there are a lot of subclasses that depend on this `_make_post_request` function. is there any concern on returning `None` here..? should this return an empty dict (`{}`) instead?

Reply:
I just searched quickly and it looks like this change would be safe

Reply:
please just update the docstring to include `None` as a possible return value

Reply:
Yeah, I searched around, looks the callers will return earlier when the first return value is `False` and they won't check second return value eventually. This change would be safe. Will update the comment.